### PR TITLE
Fixes #139 Exception when creating new table formula

### LINKED ISSUE #139: Exception when creating new table formula
1. Open NewPassiveTransportBB under Passive Transports
1. On Kinetic of NewPassiveTransport select NewTableFormula if not already selected
1. Use ellipsis buttons to try to select 'Path to table object'

Exception trace

```
Application:
MoBi 7.2.0 - Build 0

Object reference not set to an instance of an object.

Stack trace:
at MoBi.Presentation.Presenter.SelectReferencePresenterFactory.ReferenceAtParameterFor(IContainer container) in C:\git\MoBi\src\MoBi.Presentation\Presenter\SelectReferencePresenterFactory.cs:line 25
   at MoBi.Presentation.Presenter.EditTableFormulaWithOffSetFormulaPresenter.selectFormulaUseablePath(FormulaUsablePathDTO initPath, Func`2 predicate, String caption) in C:\git\MoBi\src\MoBi.Presentation\Presenter\EditTableFormulaWithOffSetFormulaPresenter.cs:line 86
   at MoBi.Presentation.Presenter.EditTableFormulaWithOffSetFormulaPresenter.SetTableObjectPath(FormulaUsablePathDTO formulaUsablePath) in C:\git\MoBi\src\MoBi.Presentation\Presenter\EditTableFormulaWithOffSetFormulaPresenter.cs:line 69
   at MoBi.UI.Views.EditTableFormulaWithOffSetFormulaView.btEditTableObjectPathOnButtonClick() in C:\git\MoBi\src\MoBi.UI\Views\EditTableFormulaWithOffSetFormulaView.cs:line 50
   at OSPSuite.UI.Controls.BaseUserControl.<>c__DisplayClass25_0.<OnEvent>b__0() in C:\projects\ospsuite-core\src\OSPSuite.UI\Controls\BaseUserControl.cs:line 112
   at OSPSuite.Utility.Extensions.ExceptionExtensions.DoWithinExceptionHandler(Object callerObject, Action actionToExecute) in C:\projects\ospsuite-utility\src\OSPSuite.Utility\Extensions\ExceptionExtensions.cs:line 16
```

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/MoBi.Presentation/Presenter/EditPassiveTransportBuildingBlockPresenter.cs [hunk 1]
```diff
@@ -48,6 +45,8 @@ public class EditPassiveTransportBuildingBlockPresenter : EditBuildingBlockPrese
          _viewItemContextMenuFactory = viewItemContextMenuFactory;
          _editTransportBuilderPresenter = editTransportBuilderPresenter;
          _transportBuilderToDTOTransportBuilderMapper = transportBuilderToDTOTransportBuilderMapper;
+         _editTransportBuilderPresenter.RemoveFormulaType<TableFormulaWithOffset>();
```
Comment (msevestre) [inline src/MoBi.Presentation/Presenter/EditPassiveTransportBuildingBlockPresenter.cs]: I do not think that TableFormulaWithOffset or SumFormula makes sense for any Transport type
It definitely does not make sense for PassiveTransport.
@Yuri05 If you agree, I would like to move this condition to the TransportBuilderPresenter unless you see a use of `TableFormulaWithOffset` or `SumFormula`...

File: src/MoBi.Presentation/Presenter/EditTableFormulaWithOffSetFormulaPresenter.cs [hunk 2]
```diff
@@ -1,90 +1,92 @@
...
+using OSPSuite.Core.Domain.UnitSystem;
 using OSPSuite.Core.Services;
 using OSPSuite.Presentation.Core;
 
 namespace MoBi.Presentation.Presenter
 {
    public interface IEditTableFormulaWithOffSetFormulaPresenter : IEditTypedFormulaPresenter
    {
-      void SetOffsetFormulaPath(FormulaUsablePathDTO formulaUsablePath);
-      void SetTableObjectPath(FormulaUsablePathDTO formulaUsablePath);
```
Comment (msevestre) [inline src/MoBi.Presentation/Presenter/EditTableFormulaWithOffSetFormulaPresenter.cs]: TELL DON'T ASK. The presenter knows what's being edited

File: src/MoBi.Presentation/Presenter/EditTableFormulaWithOffSetFormulaPresenter.cs [hunk 3]
```diff
@@ -1,90 +1,92 @@
...
+         Edit(_formula);
+      }
+
+      public void SetOffsetFormulaPath()
+      {
+         var path = selectFormulaUseablePath(_tableFormulaWithOffsetDTO.OffsetObjectPath, isValidOffsetObject, AppConstants.Captions.OffsetObjectPath);
          if (path == null) return;
 
          AddCommand(_moBiFormulaTask.ChangeOffsetObject(_formula, path, BuildingBlock));
-         _view.ShowOffsetObjectPath(_formulaUsablePathMapper.MapFrom(path, _formula));
```
Comment (msevestre) [inline src/MoBi.Presentation/Presenter/EditTableFormulaWithOffSetFormulaPresenter.cs]: view should be dumb and only have one method to bind to the object. That's it

File: tests/MoBi.Tests/Presentation/EditTableFormulaWithOffSetFormulaPresenterSpecs.cs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Core.Services;
+using OSPSuite.Presentation.Core;
+
+namespace MoBi.Presentation
+{
+   public abstract class concern_for_EditTableFormulaWithOffSetFormulaPresenter : ContextSpecification<IEditTableFormulaWithOffSetFormulaPresenter>
```
Comment (msevestre) [inline tests/MoBi.Tests/Presentation/EditTableFormulaWithOffSetFormulaPresenterSpecs.cs]: Some more tests here. Part of more tests package

Comment (msevestre) [inline tests/MoBi.Tests/Presentation/EditTableFormulaWithOffSetFormulaPresenterSpecs.cs]: @Yuri05 Could you review that PR: Especially, where should we allow TableFormulaWithOffset and SumFormula ?

**Comment by msevestre** [comment]
@Yuri05 I really need review on that now. The PR does not merge anymore so I will fix that but I have questions that need to be answered. Thanks 👍

File: src/MoBi.Presentation/Presenter/EditPassiveTransportBuildingBlockPresenter.cs (hunk 1 above)
Comment (Yuri05) [inline src/MoBi.Presentation/Presenter/EditPassiveTransportBuildingBlockPresenter.cs]: Should be OK.
If for some reason we will need those formulas for a transport, we could still overcome this by introducing transport rate as parameter, which would have **SumFormula** or **TableFormulaWithOffset**

**Pull request merged**